app/questionnaire/router.py

```diff
@@ -518,15 +518,32 @@ def get_enabled_section_keys(
         self,
     ) -> Generator[ProgressKeyType, None, None]:
         for section_id in self.enabled_section_ids:
-            repeating_list = self._schema.get_repeating_list_for_section(section_id)
-
-            if repeating_list:
+            if repeating_list := self._schema.get_repeating_list_for_section(
+                section_id
+            ):
                 for list_item_id in self._list_store[repeating_list]:
-                    section_key: ProgressKeyType = (section_id, list_item_id)
-                    yield section_key
+                    yield section_id, list_item_id
+
+            elif (
+                repeating_blocks_list := self._schema.get_repeating_blocks_list_name_for_section(
+                    section_id
+                )
+            ) and (
+                # First list collector must be present at this point
+                self._schema.get_first_list_collector_for_section(section_id).get(  # type: ignore
+                    "type"
+                )
+                == "ListCollectorContent"
+            ):
```

Comment:
If I'm not mistaken, I have a feeling this might be a misuse to fix the issue mentioned in https://github.com/ONSdigital/eq-questionnaire-runner/pull/1156#discussion_r1263756040.

Reply:
Removed now, thanks! It wasn't needed.

Reply:
Removed the whole check now as wasn't required given the other changes 👍